src/silverlabnwb/signature.py

```diff
@@ -50,6 +55,34 @@ def __init__(self):
             ('/', 'nwb_version'),
         ]:
             self.ignore_attribute(path, attr)
+        # some datasets need platform-specific correction of type
+        self._cast_paths = {}
+        for dataset_path, expected, corrected in [
+            ('/acquisition/EyeCam/dimension', int32, int64),
+            ('/acquisition/WhiskersCam/dimension', int32, int64),
+            ('/acquisition/Zstack_.*/dimension', int32, int64),
+            ('/intervals/epochs/id', int32, int64),
+            ('/intervals/trials/id', int32, int64),
```

Comment:
This list could be shortened by grouping together some paths, but I'm not sure whether this is clearer:
```suggestion
            ('/intervals/(epochs|trials)/id', int32, int64),
```
(and the same with the `colnames` attribute  below)